infrastructure/terraform/modules/logicapp/logicapp_apiconnections.tf

```diff
@@ -39,3 +39,54 @@ resource "azapi_resource" "api_connection_arm" {
   schema_validation_enabled = false
   response_export_values    = ["properties.connectionRuntimeUrl"]
 }
+
+resource "azurerm_resource_group_template_deployment" "api_connection_arm_access_policy" {
+  for_each = var.logic_app_api_connections
+
+  name                = "${var.logic_app_name}-api-${each.key}"
+  resource_group_name = var.resource_group_name
+  deployment_mode     = "Incremental"
+
+  template_content = <<TEMPLATE
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "connectionName": {
+      "type": "string"
+    },
+    "principalId": {
+      "type": "string"
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Web/connections/accessPolicies",
+      "name": "[concat(parameters('connectionName'), '/', parameters('principalId'))]",
+      "apiVersion": "2016-06-01",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "principal": {
+          "type": "ActiveDirectory",
+          "identity": {
+            "objectId": "[parameters('principalId')]",
+            "tenantId": "[subscription().tenantId]"
+          }
+        }
+      }
+    }
+  ]
+}
+TEMPLATE
+
+  parameters_content = jsonencode(
+    {
+      "connectionName" = {
+        value = "${var.logic_app_name}-api-${each.key}"
+      },
+      "principalId" = {
+        value = "${azurerm_logic_app_standard.logic_app_standard.identity[0].principal_id}"
+      }
+    }
+  )
+}
```

Comment:
I have done the same using azapi. If you don't mind let's use azapi as ARM templates usually create some challenges from a plan perspective.

Reply:
This PR now includes the `azapi` resource: https://github.com/Azure/ai-hub/pull/166